### DISCUSSION TITLE
Jetson Nano: `apt install` can't find ROS packages

### DISCUSSION THREAD

**Discussion by c1505** [iRobotEducation/create3_docs#303] [Questions]
The instructions here: https://iroboteducation.github.io/create3_docs/setup/jetson/ have you run this command
```
sudo apt update && sudo apt install -y ros-galactic-ros-base python3-colcon-common-extensions python3-rosdep ros-galactic-rmw-fastrtps-cpp ros-galactic-rmw-cyclonedds-cpp ros-galactic-irobot-create-msgs
```
None of the packages can be found.  Tried to additional package sources without luck.  

This issue was created last week with the same problem, but it was then closed without a solution https://github.com/iRobotEducation/create3_docs/issues/288

**Comment by alsora** [comment]
Hi @c1505, could you please show the error that you get?

In any case, it may be related to not have setup your debian package sources.
You can try the following
```bash
sudo apt install software-properties-common
sudo add-apt-repository universe
sudo apt update && sudo apt install curl
sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
sudo apt update
```


And then try again the install command.

See here for more details: https://docs.ros.org/en/galactic/Installation/Ubuntu-Install-Debians.html#setup-sources

**Comment by tokk-nv** [comment]
Hi @c1505,

Did you run the `apt install` commands inside the container?
If not, please try running the command inside the container by running that after the docker run command mentioned in [the former sub-section](https://iroboteducation.github.io/create3_docs/setup/jetson/#start-ros-2-galactic-container).
```
git clone https://github.com/dusty-nv/jetson-containers/
cd jetson-containers
scripts/docker_run.sh -c dustynv/ros:galactic-ros-base-l4t-r32.6.1
```

Jetson Nano runs JetPack 4.x which is based on Ubuntu 18.04.
So if you try to install Galactic packages from apt, Jetson Nano doesn’t have those available because it’s on Ubuntu 18.04 and Galactic apt/Debian packages are only available for 20.04.
Hence NVIDIA built those from sources in the ROS Galactic containers (e.g. `ros:galactic-ros-base-l4t-r32.6.1`).
